drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+
+    # initialize the Drakvuf capa feature extractor with the captured calls
+    extractor = get_drakvuf_feature_extractor(calls)
+
+    # apply a filter to the extractor if we wish to only analyze a specific set of processes
+    if malware_pids:
+        extractor = capa.features.extractors.base_extractor.ProcessFilter(
+            extractor, malware_pids
+        )
+
+    # extract dymamic capabilities
+    capabilities, _ = capa.capabilities.common.find_capabilities(rules, extractor)
+
+    # analysis_dir is returned for conformity reasons with the ttp generation function (same routine for both static and dynamic capability extraction)
+    return analysis_dir, capabilities
+
+
+def get_process_memory_dumps(analysis_dir: Path, pid: int) -> Iterator[str]:
+    # get the memory dumps from a specific process.
+    # this is used mainly to get the memdumps by a malware process
+    with (analysis_dir / "memdump.log").open("r") as f:
+        for line in f:
+            dump = orjson.loads(line)
+            if dump["PID"] == pid:
+                yield dump["DumpFilename"]
+
+
+def static_capa_analysis(
+    dump_path: Path, rules: capa.rules.RuleSet
+) -> Tuple[Path, capa.render.result_document.MatchResults]:
+
+    # get the input file's capa format
+    try:
+        input_format = capa.helpers.get_auto_format(dump_path)
+    except capa.main.UnsupportedFormatError:
+        logger.debug("dump %s has an unsupported format", dump_path)
+        return
+
+    # get the input file's adequate analysis backend
+    if input_format == capa.main.FORMAT_DOTNET:
+        backend = capa.main.BACKEND_DOTNET
+    else:
+        backend = capa.main.BACKEND_VIV
+
+    # get the right capa feature extractor for the input: Vivisect, .NET, etc.
+    extractor = capa.loader.get_extractor(
+        dump_path,
+        input_format,
+        capa.main.OS_WINDOWS,
+        backend,
+        [],
+        False,
+        False,
+        dump_path,
+    )
+
+    # extract capabilities from the file, and ignore the returned metadata
+    capabilities, _ = capa.capabilities.common.find_capabilities(
+        rules, extractor, disable_progress=True
+    )
+
+    # return the capabilities alongside the memdump's path (for TTP categorization purposes)
+    return dump_path, capabilities
+
+
+def static_memory_dumps_capa_analysis(
+    analysis_dir: Path, rules: capa.rules.RuleSet, malware_pids: List[int] = []
+) -> Iterator[Tuple[Path, capa.render.result_document.MatchResults]]:
+    malware_dumps = list(
+        itertools.chain(
+            *(get_process_memory_dumps(analysis_dir, pid) for pid in malware_pids)
+        )
+    )
+
+    # extract all memory dumps temporarily into a dumps/ folder
+    with zipfile.ZipFile(analysis_dir / "dumps.zip", "r") as zip_ref:
+        try:
+            # try to extract the memory dumps into the analysis folder
+            zip_ref.extractall(analysis_dir)
+            dumps = analysis_dir / "dumps"
```

Comment:
No need to implement two methods. I think we can always extract to /tmp and remove part in try: